### PR TITLE
fix docs pyfunc.log_model

### LINKED ISSUE #9550: [DOC-FIX] mlflow.pyfunc.log_model example uses mlflow.pyfunc.save_model
### Willingness to contribute

Yes. I would be willing to contribute a document fix with guidance from the MLflow community.

### URL(s) with the issue

[https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=pyfunc%20log_model#mlflow.pyfunc.log_model](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=pyfunc%20log_model#mlflow.pyfunc.log_model)

### Description of proposal (what needs changing)

[This code example](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=pyfunc%20log_model#mlflow.pyfunc.log_model) in the documentation should explain how to use `pyfunc.log_model()` but shows `pyfunc.save_model()`

```python
from typing import List, Dict
import mlflow


class MyModel(mlflow.pyfunc.PythonModel):
    def predict(self, context, model_input: List[str], params=None) -> List[str]:
        return [i.upper() for i in model_input]


mlflow.pyfunc.save_model("model", python_model=MyModel(), input_example=["a"])
model = mlflow.pyfunc.load_model("model")
print(model.predict(["a", "b", "c"]))  # -> ["A", "B", "C"]
```

A draft example would be something in the lines of (cfr: [mlflow.pyfunc example](https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=pyfunc%20log_model#mlflow.pyfunc.PyFuncModel))

```python

# define a custom model
class MyModel(mlflow.pyfunc.PythonModel):
    def predict(self, context, model_input, params=None):
        return self.my_custom_function(model_input, params)

    def my_custom_function(self, model_input, params=None):
        # do something with the model input
        return 0

# save the model
my_model = MyModel()
with mlflow.start_run():
    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model)
```

### PR DESCRIPTION
**Pull request by Beramos**
## Related Issues/PRs

Resolve #9550 

## What changes are proposed in this pull request?
Update `pyfunc.log_model()` examples to better reflect `log_model()` and `save_model()`-functionality

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

I've build the documentation and verified the output. I've also ran the examples to make sure they were valid. There seems to be no relevant unit tests for this change.

## Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [ ] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Update `pyfunc.log_model()` examples to better reflect `log_model()` and `save_model()`-functionality

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pyfunc/__init__.py [hunk 1]
```diff
@@ -1926,9 +1926,13 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                     return [i.upper() for i in model_input]
 
 
-            mlflow.pyfunc.save_model("model", python_model=MyModel(), input_example=["a"])
-            model = mlflow.pyfunc.load_model("model")
-            print(model.predict(["a", "b", "c"]))  # -> ["A", "B", "C"]
+            my_model = MyModel()
+            with mlflow.start_run():
+                model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model)
```
Comment (serena-ruan) [inline mlflow/pyfunc/__init__.py:1931]:
```suggestion
            with mlflow.start_run():
                model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=MyModel(), input_example=["a"])
```
This extra variable definition seems useless, and we can still keep input_example :)

**Review by serena-ruan — APPROVED** [review]
LGTM with one small nit! BTW could you sign off your commit as it's a requirement for passing CI? You could update by running `git rebase HEAD~2 --signoff`

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (Beramos) [reply mlflow/pyfunc/__init__.py:1931]: There are two remarks here, the extra variable definition is indeed useless. But I stole it from [another documentation section](https://github.com/mlflow/mlflow/blob/19079d163a7569b759759e6cc2518bb18f62ce1d/mlflow/pyfunc/__init__.py#L498).

using the `input_example`-arg raises an error. So this needs to be fixed too, that is why I left it out.

```
Traceback (most recent call last):
  File "/home/bramdj/anaconda3/envs/mlflow-dev-env/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/bramdj/anaconda3/envs/mlflow-dev-env/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/home/bramdj/Desktop/temp$.py", line 12, in <module>
    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model, input_example=["a"])
  File "/home/bramdj/Desktop/git/mlflow/mlflow/pyfunc/__init__.py", line 2024, in log_model
    return Model.log(
  File "/home/bramdj/Desktop/git/mlflow/mlflow/models/model.py", line 579, in log
    flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
  File "/home/bramdj/Desktop/git/mlflow/mlflow/pyfunc/__init__.py", line 1819, in save_model
    if signature := _infer_signature_from_type_hints(
  File "/home/bramdj/Desktop/git/mlflow/mlflow/models/signature.py", line 310, in _infer_signature_from_type_hints
    output_example = func(input_example) if input_example else None
TypeError: predict() missing 1 required positional argument: 'model_input'
``

Comment (Beramos) [reply mlflow/pyfunc/__init__.py]: I've made some changes to remove the extra variable definition, if you want I can add `input_example` too but that is up to you

Comment (Beramos) [reply mlflow/pyfunc/__init__.py]: DCO check passes to, I was apparently not consistent in my sign offs. My apologies

**Comment by harupy** [comment]
@mlflow-automation autoformat

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (serena-ruan) [reply mlflow/pyfunc/__init__.py]: Oh this sounds like a problem... We should definitely fix it.
In that case, feel free to remove the input_example in this docs, we can add it back later if needed. Thanks!

Comment (Beramos) [reply mlflow/pyfunc/__init__.py]: If I have time to dig up some more information, I'll open a new issue for that.

Comment (Beramos) [reply mlflow/pyfunc/__init__.py]: @serena-ruan, I have signed off on everything and made the requested changes, feel free to merge when all checks pass.

**Comment by Beramos** [comment]
@serena-ruan, it seems that one of the actions failed, but quite unrelated to my changes.

File: mlflow/pyfunc/__init__.py (hunk 1 above)
Comment (serena-ruan) [reply mlflow/pyfunc/__init__.py]: Thank you!

**Pull request merged**